app/models/event_schedule.rb

```diff
@@ -37,6 +39,16 @@ def replacement?
 
   private
 
+  def start_after_end_hour
+    errors.
+    add(:start_time, "can't be after the conference end hour (#{event.program.conference.end_hour})") if event.program && event.program.conference && event.program.conference.end_hour && start_time && start_time.strftime('%H').to_i > event.program.conference.end_hour
```

Comment:
For this complicated conditional I don't like using the conditional at the end of the line... I would go with
```
if ...
...
end
```
Easier to read in my opinion...